src/jit/gentree.cpp

```diff
@@ -6746,42 +7009,36 @@ void GenTreeIntCon::FixupInitBlkValue(var_types asgType)
 
 // Initializes a BlkOp GenTree
 // Preconditions:
-//     - Result is a GenTreeBlkOp that is newly constructed by gtNewCpObjNode or gtNewBlkOpNode
+//     - Result is an assignment that is newly constructed
 //
 // Parameters:
-//     - result is a GenTreeBlkOp node that is the node to be initialized.
-//     - oper must be either GT_INITBLK or GT_COPYBLK
+//     - result is an assignment node that is to be initialized.
 //     - dst is the target (destination) we want to either initialize or copy to
```

Comment:
Here too dst has to be either a GT_OBJ/BLK/DYN_BLK or a LclVar?  If yes, please update both the function header comment and assert the same in this routine.